redwood-compose/src/commonTest/kotlin/app/cash/redwood/compose/BackHandlerTest.kt

```diff
@@ -0,0 +1,204 @@
+        Box {
+          BackHandler {
+            backCounter += 1
+          }
+        }
+        Text(backCounter.toString())
+      }
+
+      assertThat(awaitSnapshot()).contains(TextValue(text = "0"))
+      onBackPressedDispatcher.onBackPressed()
+      assertThat(awaitSnapshot()).contains(TextValue(text = "1"))
+    }
+  }
+
+  @Test
+  fun outermostEnabledAndInnermostDisabledBackHandlers() = runTest {
+    val onBackPressedDispatcher = FakeOnBackPressedDispatcher()
+    TestSchemaTester(onBackPressedDispatcher) {
+      setContent {
+        var backCounter by remember { mutableStateOf(0) }
+        BackHandler {
+          backCounter += 1
+        }
+        Box {
+          BackHandler(enabled = false, throwingOnBack)
+        }
+        Text(backCounter.toString())
+      }
+
+      assertThat(awaitSnapshot()).contains(TextValue(text = "0"))
+      onBackPressedDispatcher.onBackPressed()
+      assertThat(awaitSnapshot()).contains(TextValue(text = "1"))
+    }
+  }
+
+  @Test
+  fun outermostDisabledAndInnermostEnabledBackHandlers() = runTest {
+    val onBackPressedDispatcher = FakeOnBackPressedDispatcher()
+    TestSchemaTester(onBackPressedDispatcher) {
+      setContent {
+        var backCounter by remember { mutableStateOf(0) }
+        BackHandler(enabled = false, throwingOnBack)
+        Box {
+          BackHandler {
+            backCounter += 1
+          }
+        }
+        Text(backCounter.toString())
+      }
+
+      assertThat(awaitSnapshot()).contains(TextValue(text = "0"))
+      onBackPressedDispatcher.onBackPressed()
+      assertThat(awaitSnapshot()).contains(TextValue(text = "1"))
+    }
+  }
+
+  @Test
+  fun outermostDisabledAndInnermostDisabledBackHandlers() = runTest {
+    val onBackPressedDispatcher = FakeOnBackPressedDispatcher()
+    TestSchemaTester(onBackPressedDispatcher) {
+      setContent {
+        val backCounter by remember { mutableStateOf(0) }
+        BackHandler(enabled = false, throwingOnBack)
+        Box {
+          BackHandler(enabled = false, throwingOnBack)
+        }
+        Text(backCounter.toString())
+      }
+
+      assertThat(awaitSnapshot()).contains(TextValue(text = "0"))
+      onBackPressedDispatcher.onBackPressed()
+      assertNoSnapshot()
+    }
+  }
+}
+
+private suspend fun TestRedwoodComposition<List<WidgetValue>>.assertNoSnapshot() {
+  assertFailure { awaitSnapshot() }.isInstanceOf<TimeoutCancellationException>()
+}
+
+private class FakeOnBackPressedDispatcher : OnBackPressedDispatcher {
+  private val onBackPressedCallbacks = ArrayDeque<OnBackPressedCallback>()
+
+  override fun addCallback(onBackPressedCallback: OnBackPressedCallback): Cancellable {
+    onBackPressedCallbacks += onBackPressedCallback
+    return object : Cancellable {
+      override fun cancel() {
+        onBackPressedCallbacks -= onBackPressedCallback
+      }
+    }
+  }
+
+  fun onBackPressed() {
+    val callbackToNotify = onBackPressedCallbacks.lastOrNull { it.isEnabled }
```

Comment:
Because this fake implements the behavior that chooses the last callback to invoke, the tests which put multiple back handlers into the composition (at least when both are enabled) are exercising this behavior and not any actual library behavior.

Reply:
Good point. Just updated the PR to inline this function and add the desired assertions.